chore(sinks, codecs): Remove legacy `EncodingConfiguration` (#13518)
vectordotdev/vector: website/content/en/highlights/2022-08-16-0-24-0-upgrade-guide.md

   the `native` codec) make sure you upgrade the consumers first followed by the
   producers to ensure newer versions of Vector aren't sending data to older
   versions, which may not be able to be read.

#### `codec` field on sink `encoding` must be specified explicitly {#sink-encoding-codec}

Setting the `encoding` value on a sink by a string has been deprecated in

> `0.23.0` and will be removed in this release. To migrate have a look at the

Reviewer: ```suggestion
`0.23.0` and is removed in this release. To migrate have a look at the
```

> [upgrade guide for deprecated shorthand values for `encoding` options](https://vector.dev/highlights/2022-07-07-0-23-0-upgrade-guide/#deprecated-encoding-shorthand).

Reviewer: Generally for links we avoid including the domain so that branch deploys work correctly. Same for the below link.
```suggestion
[upgrade guide for deprecated shorthand values for `encoding` options](/highlights/2022-07-07-0-23-0-upgrade-guide/#deprecated-encoding-shorthand).
```

#### `ndjson` on sink `encoding` is now `json` encoding + `newline_delimited` framing {#sink-encoding-json}

> The `ndjson` sink `encoding` value has been deprecated in `0.23.0` and will be
> removed in this release. To migrate have a look at the

Reviewer: ```suggestion
The `ndjson` sink `encoding` value has been deprecated in `0.23.0` and is
removed in this release. To migrate have a look at the
```

#### `codec` field on sink `encoding` must be specified explicitly {#sink-encoding-codec}

Setting the `encoding` value on a sink by a string has been deprecated in
`0.23.0` and is removed in this release. To migrate have a look at the
[upgrade guide for deprecated shorthand values for `encoding` options](/highlights/2022-07-07-0-23-0-upgrade-guide/#deprecated-encoding-shorthand).

#### `ndjson` on sink `encoding` is now `json` encoding + `newline_delimited` framing {#sink-encoding-json}

The `ndjson` sink `encoding` value has been deprecated in `0.23.0` and is
removed in this release. To migrate have a look at the

> [upgrade guide for the `ndjson` sink `encoding` value](https://vector.dev/highlights/2022-07-07-0-23-0-upgrade-guide/#sink-encoding-ndjson-json).

Author: Unfortunately the GitHub UI didn't let me do it while applying your suggestions. There was another link in the document that I updated as well 👍
